core/src/crds_gossip_push.rs

```diff
@@ -153,17 +162,29 @@ impl CrdsGossipPush {
         let value_hash = new_value.value_hash;
         if let Some((_, ref mut received_set)) = self.received_cache.get_mut(&value_hash) {
             received_set.insert(*from);
-            return Err(CrdsGossipError::PushMessageAlreadyReceived);
+            self.num_dups += 1;
+            return Ok(Some(new_value));
         }
         let old = crds.insert_versioned(new_value);
         if old.is_err() {
+            self.num_old += 1;
             return Err(CrdsGossipError::PushMessageOldVersion);
         }
         let mut received_set = HashSet::new();
         received_set.insert(*from);
```

Comment:
Do you think we should only insert into the received set if the message comes from a non-pruned node?  Want to avoid something like: 

B sends message to A
C sends message to A
A prunes C
C sends message to A (before C receives the prune)
B sends message to A
A prunes B